src/shims/tls.rs

```diff
@@ -74,38 +79,62 @@ impl<'tcx> TlsData<'tcx> {
     pub fn load_tls(
         &self,
         key: TlsKey,
+        thread_id: ThreadId,
         cx: &impl HasDataLayout,
     ) -> InterpResult<'tcx, Scalar<Tag>> {
         match self.keys.get(&key) {
-            Some(&TlsEntry { data, .. }) => {
-                trace!("TLS key {} loaded: {:?}", key, data);
-                Ok(data.unwrap_or_else(|| Scalar::null_ptr(cx).into()))
+            Some(TlsEntry { data, .. }) => {
+                let value = data.get(&thread_id).cloned();
```

Comment:
```suggestion
                let value = data.get(&thread_id).copied();
```
I think this should work, `Scalar` are `Copy`.